File: packages/shared/src/types/user-business.ts
import { type BaseModel } from "./base-model-type";
import { type IBusinessVerification } from "./business-verification";
import { type ICollection } from "./collection";

export interface IUserBusiness extends BaseModel {
  user_id: string;
  collection_id:string | null;
  name: string;
  type: string;
  phone: string;
  street_address: string;
  city: string;
  state: string;
  country: string;

  verification?: IBusinessVerification;
}


export interface IUserBussinessWithCollection extends IUserBusiness {
  collection: ICollection[]
}

File: apps/spine/lib/drift/model/user_business.dart
import 'package:drift/drift.dart';
import 'package:spine/drift/model/base.dart';
import 'package:spine/drift/model/business_verification.model.dart';


class UserBusiness extends Table with TableMixin {
    late final userId = text()();
    late final collectionId = text()();
    late final name = text()();
    late final type = text()();
    late final phone = text()();
    late final streetAddress = text()();
    late final city = text()();
    late final state = text()();
    late final country = text()();

    
    late final verification = text().references(BusinessVerification, #id)();

}
